### PR TITLE
Add support for cross-validation

### PR DESCRIPTION
**Pull request by FlyingPumba**

### CODE REVIEW

File: cot_probing/attn_probes_data_proc.py [hunk 1]
```diff
@@ -115,41 +115,65 @@ def split_data(
...
     assert isinstance(batch_size, int)
-    setup_determinism(data_seed)
+
+    # Get fold information
+    fold = data_loading_kwargs.get("fold", 0)
+    n_folds = data_loading_kwargs.get("n_folds", 1)
+    assert isinstance(fold, int) and isinstance(n_folds, int)
+    assert 0 <= fold < n_folds
+
+    setup_determinism(train_seed)
```
Comment (jettjaniak) [inline cot_probing/attn_probes_data_proc.py:130]: This should be cv seed?

Comment (FlyingPumba) [reply cot_probing/attn_probes_data_proc.py:130]: I thought we were going to use the train seed for shuffling the data after the fold split. The cv is used for shuffling before the fold split.

**Comment by jettjaniak** [comment]
Quite proud of that one.
All config required for training is in TrainerConfig (prev. ProbingConfig). Instead of data_loading_kwargs and DatasetConfig we have single DataConfig, which is a sub of TrainerConfig. Also from_wandb() shouldn't break anymore when we change config structure, as we don't need to manually specify parameter names - TrainerConfig is equivalent to a run config (so no more redundancy).
experiment_uuid is shared between folds, so when we want to analyze a single training seed, single layer etc. but for all folds, we look at a single experiment_uuid
We have a test that trains a dummy probe and then loads it from wandb.
Unfortunately we don't have tests for anything else, so I've probably broken a lot of stuff. Commit [f249d6e](https://github.com/jettjaniak/cot-probing/pull/9/commits/f249d6e6401f755a759a85d9a3a87ff6c1eb53bf) has changes from the automatic refactors. Many of these files probably will need to be re-written anyway to work with k-fold cv anyway.

**Comment by jettjaniak** [comment]
running
```
for LAYER in {10..19} {00..09} {20..32}; do for FSP in biased no; do for SEED in {1..10}; do for PC in tied untied; do ./scripts/train.py -f activations/acts_L${LAYER}_${FSP}-fsp_oct28-1156.pkl --pc $PC --cvs 0 --ts $SEED --wandb-project cross-val-unbal; done; done; done; done;
```

**Pull request merged**